src/middlewared/middlewared/plugins/enclosure_/enclosure_class.py

```diff
@@ -306,16 +312,44 @@ def _parse_elements(self, elements):
                 # does this enclosure slot support identification? (i.e. lighting up LED)
                 parsed[SUPPORTS_IDENTIFY_KEY] = self.disks_map[slot][SUPPORTS_IDENTIFY_KEY]
 
+                # does this enclosure slot support reporting identification status?
+                # (i.e. whether the LED is currently lit up)
+                light_status = self.disks_map[slot].get(SUPPORTS_IDENTIFY_STATUS_KEY, parsed[SUPPORTS_IDENTIFY_KEY])
+                if light_status:
+                    # Per SES-4-r3 Table 84 — Array Device Slot status element
+                    # The IDENT bit is Bit #1 of Byte #2.  These have been flattened
+                    # above into value_raw.
+                    if value_raw & 0x0200:
+                        parsed[DRIVE_BAY_LIGHT_STATUS] = 'ON'
+                    else:
+                        parsed[DRIVE_BAY_LIGHT_STATUS] = 'OFF'
+                else:
+                    parsed[DRIVE_BAY_LIGHT_STATUS] = None
+
                 mapped_slot = self.disks_map[slot][MAPPED_SLOT_KEY]
                 # is this a front, rear or internal slot?
                 parsed.update(disk_position_mapping.get(mapped_slot, dict()))
 
+                # Pick out the original slot number from the descriptor,
+                # for example 'slot00', 'slot01', ... etc
+                try:
```

Comment:
Sorry but this won't work. I'll go investigate why we're off by 1 for SES proper devices.

Reply:
We're off by one in py-sg3 in the `status()` method here
```
                elements = {}
                for i in range(num_dsc):
                    type = tc_buff[i]
                    dsc = self.decode_response(desc_buff[i])
                    stts = [int(x, 16) for x in self.decode_response(stat_buff[i]).split()]
                    elements[i] = { "type": type, "descriptor": dsc, "status": stts }